File: py_cli/xahseelqjgcx.py
import requests
import json
import os

def clear_screen():
    os.system('cls' if os.name == 'nt' else 'clear')

def print_data(explanation, data, key):
    print(f"{explanation}：{data.get(key, 'N/A')}")

def main():
    while True:
        print("西安市城六区中考录取结果API调用器 Python版 V2")
        input0 = input("请输入查询年份: ")
        input1 = input("请输入准考证号: ")
        input2 = input("请输入身份证号: ")

        url = f"http://222.91.162.190:7173/OperatorData.ashx?Type=MTC&YearID={input0}&Zkzh={input1}&Code=&r=100&Sfzh={input2}"

        try:
            response = requests.get(url)
            response.raise_for_status()
            data = response.json()

            # Outputdata
            print("查询成功，以下为清单数据：")
            print_data("身份证号", data, "sfzh")
            print_data("姓名", data, "xm")
            print_data("毕业学校代码", data, "byxxdm")
            print_data("录取批次", data, "lqpc")
            print_data("录取学校类型", data, "clqpc")
            print_data("录取类别", data, "lqlb")
            print_data("录取类别文字", data, "clqlb")
            print_data("录取学校代码", data, "xxdm")
            print_data("录取学校", data, "xxmc")
            print_data("查询用时", data, "totaltime")
            print("以上为清单数据。如录取学校值为空，则暂无录取信息。请核对信息是否正确。如果信息无误，请耐心等待新一批的录取。")
            print("This is a Free and Open Source Software (GPL-3.0).")
            print("Made by B5-Software")

        except requests.exceptions.RequestException as e:
            print(f"请求失败: {e}")
        except json.JSONDecodeError:
            print("响应格式错误")

        input("按回车键继续查询……")
        clear_screen()

if __name__ == "__main__":
    main()

File: cpp_cli/xahseelqjgcx.cpp
#include <iostream>
#include <string>
#include <curl/curl.h>
#include <json/json.h>
#include <conio.h> // for _getch()

// W-String
size_t WriteCallback(void* contents, size_t size, size_t nmemb, void* userp) {
    ((std::string*)userp)->append((char*)contents, size * nmemb);
    return size * nmemb;
}

// 输出
void printData(const std::string& explanation, const Json::Value& data, const std::string& key) {
    std::cout << explanation << "：" << data[key].asString() << std::endl;
}

// 清屏
void clearScreen() {
    #ifdef _WIN32
        system("cls");
    #else
        system("clear");
    #endif
}

int main() {
    while (true) {
        std::string input0, input1, input2;
        
        std::cout << "西安市城六区中考录取结果API调用器 CPP版 V2" << std::endl;

        std::cout << "请输入查询年份: ";
        std::getline(std::cin, input0);

        std::cout << "请输入准考证号: ";
        std::getline(std::cin, input1);

        std::cout << "请输入身份证号: ";
        std::getline(std::cin, input2);

        std::string url = "http://222.91.162.190:7173/OperatorData.ashx?Type=MTC&YearID=" + input0 +
                          "&Zkzh=" + input1 + "&Code=&r=100&Sfzh=" + input2;

        CURL* curl;
        CURLcode res;
        std::string readBuffer;

        curl = curl_easy_init();
        if(curl) {
            curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
            curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
            curl_easy_setopt(curl, CURLOPT_WRITEDATA, &readBuffer);
            res = curl_easy_perform(curl);
            curl_easy_cleanup(curl);

            if(res != CURLE_OK) {
                std::cerr << "curl_easy_perform() failed: " << curl_easy_strerror(res) << std::endl;
            } else {
                // 解析
                Json::Reader reader;
                Json::Value obj;
                reader.parse(readBuffer, obj);

                // 输出
                printData("身份证号", obj, "sfzh");
                printData("姓名", obj, "xm");
                printData("毕业学校代码", obj, "byxxdm");
                printData("录取批次", obj, "lqpc");
                printData("录取学校类型", obj, "clqpc");
                printData("录取类别", obj, "lqlb");
                printData("录取类别文字", obj, "clqlb");
                printData("录取学校代码", obj, "xxdm");
                printData("录取学校", obj, "xxmc");
                printData("查询用时", obj, "totaltime");
            }
        }

        std::cout << "按任意键继续查询..." << std::endl;
        _getch();
        clearScreen();
    }

    return 0;
}
